hazelcast/src/main/java/com/hazelcast/internal/util/ServiceLoader.java

```diff
@@ -50,8 +51,8 @@
  * environments like application or OSGi servers.
  */
 public final class ServiceLoader {
-    //compatibility flag to re-introduce behaviour from 3.8.0 with classloading fallbacks
-    private static final boolean USE_CLASSLOADING_FALLBACK = getBoolean("hazelcast.compat.classloading.hooks.fallback");
+    // kill-switch for URLDefinition#equals fix to take into account classloader
+    private static final boolean URLDEFINITION_COMPAT = getBoolean("hazelcast.compat.urldefinition");
```

Comment:
consider changing this to `hazelcast.compat.classloading.urldefinition` 

If someone ends-up using it (hopefully not!) then it would be good have a switch which indicates it's somehow related to classloading. it will be probably appended to a list of JVM options.